website/oauth/utils.py

```diff
@@ -6,3 +8,8 @@
 def get_service(name):
     """Given a service name, return the provider class"""
     return PROVIDER_LOOKUP[name]()
+
+def get_available_scopes():
+    scopes = [(scope_key, public_scopes[scope_key].description) for scope_key in public_scopes if public_scopes[scope_key].is_public]
```

Comment:
Since this is a dictionary, you can make this more concise using iteritems. Below is one sample rewrite (that gives mako the entire namedtuple of data about the scope- if you want to stick to your simplified tuple instead you'll need to modify accordingly)

```
return sorted( (scope_name, scope_data) 
    for scope_name, scope_data in public_scopes.iteritems()
    if scope_data.is_public is True)
```